### ISSUE TITLE
Add "Host: " header (or TLS SNI)

### ISSUE THREAD

**Issue by raspbeguy** [equeim/tremotesf-android#49]
I use transmission web gui behind a ssl offloader (HAproxy) which redirect web requests to several other web backends. To dispatch traffic it uses "Host: " web header. It seems that tremotesf doesn't include it in its web requests.

EDIT: after reflexion, it's most likely lack of SNI in TLS negociation. I didn't check in details.

**Comment by equeim** [comment]
I just checked on the latest version and it does use both Host header and SNI (I tested it on lighttpd server, and also checked with Wireshark).
What does it print in logcat (with LibTremotesf tag)?

**Comment by raspbeguy** [comment]
I think I've got the solution.

The problem is that tremotesf puts the port number in the host header. For instance, its requests look like:

```
Host: transmission.example.com:443
```

That is the first time I see the port number in the host header. I was thinking that was an error from tremotesf, but apparently this is a legit syntax of the host header. So when you administrate a SSL offloader or a reverse proxy, you have basically 2 choices:

* add to every virtual host an alias containing the port number ;
* tell the reverse proxy to strip the port number before treating incoming requests.

The second requires less effort so this is what I did. For HAproxy, this means adding the following in the frontend section:

```
http-request replace-header Host ([^:]+) \1
```
